pkg/api/types.go

```diff
@@ -94,3 +97,25 @@ type MetricsCollector struct {
 	// Later, the collection can be extended to other providers.
 	Enabled bool
 }
+
+type Prometheus struct {
+	URL                string
+	AuthToken          AuthToken
+	InsecureSkipVerify bool
+}
+
+type AuthToken struct {
+	// raw for a raw authentication token
+	Raw string
+	// secretReference references an authentication token.
+	// secrets are expected to be created under the descheduler's namespace.
+	SecretReference SecretReference
+}
+
+// SecretReference holds a reference to a Secret
+type SecretReference struct {
+	// namespace is the namespace of the secret.
+	Namespace string
```

Comment:
> secrets are expected to be created under the descheduler's namespace.

Do we need the Namespace field then?

What happens if a different namespace is set? How is the validation and rbac managed?